deps/src/main/java/com/microsoft/azure/sdk/iot/deps/serializer/ErrorCodeDescription.java

```diff
@@ -0,0 +1,193 @@
+            case(400012):
+                return DeserializationError;
+            case(400013):
+                return BulkRegistryOperationFailure;
+            case (400027):
+                return ConnectionForcefullyClosedOnNewConnection;
+            case(400301):
+                return CannotRegisterModuleToModule;
+            case(401001):
+                return IotHubNotFound;
+            case(401002):
+                return IotHubUnauthorizedAccess;
+            case(401003):
+                return IotHubUnauthorized;
+            case(403001):
+                return IotHubSuspended;
+            case(403002):
+                return IotHubQuotaExceeded;
+            case(403003):
+                return JobQuotaExceeded;
+            case(403004):
+                return DeviceMaximumQueueDepthExceeded;
+            case(403005):
+                return IotHubMaxCbsTokenExceeded;
+            case(404001):
+                return DeviceNotFound;
+            case(404002):
+                return JobNotFound;
+            case(404003):
+                return PartitionNotFound;
+            case(404010):
+                return ModuleNotFound;
+            case (404103):
+                return DeviceNotOnline;
+            case (404104):
+                return DeviceConnectionClosedRemotely;
+            case(409001):
+                return DeviceAlreadyExists;
+            case(409301):
+                return ModuleAlreadyExistsOnDevice;
+            case(412001):
+                return PreconditionFailed;
+            case(412002):
+                return DeviceMessageLockLost;
+            case(413001):
+                return MessageTooLarge;
+            case(413002):
+                return TooManyDevices;
+            case(413003):
+                return TooManyModulesOnDevice;
+            case(429001):
+                return ThrottlingException;
+            case(429002):
+                return ThrottleBacklogLimitExceeded;
+            case(429003):
+                return InvalidThrottleParameter;
+            case(500001):
+                return ServerError;
+            case(500002):
+                return JobCancelled;
+            case(503001):
+                return ServiceUnavailable;
+            case (400):
+                return BadFormat;
+            case (401):
+                return Unauthorized;
+            case (403):
+                return Forbidden;
+            case (404):
+                return NotFound;
+            case (409):
+                return Conflict;
+            case (412):
+                return PreconditionFailed;
+            case (413):
+                return RequestEntityTooLarge;
+            case (429):
+                return ThrottlingException;
+            case (500):
+                return InternalServerError;
+            case (503):
+                return ServiceUnavailable;
+            default:
+                return UnclassifiedErrorCode;
```

Comment:
Have these been validated by service ? i.e. do they support the above for what do they mean?

Reply:
These are all taken directly from the C# sdk [here](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/service/src/Common/Exceptions/ErrorCode.cs)

Reply:
👍